addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -135,3 +135,60 @@ def test10_requires_backorder_multi_lines(self):
         # Check a backorder is needed
         mls = pick.move_line_ids
         self.assertTrue(pick._requires_backorder(mls))
+
+    def test11_backorder_move_lines_simple(self):
+        """ Backorder move lines for incomplete pick """
+        self.create_quant(self.fig.id, self.test_stock_location_02.id, 50)
+        self.create_quant(self.banana.id, self.test_stock_location_02.id, 50)
+        pick = self.Picking.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{'product': self.fig, 'qty': 50}, {'product': self.banana, 'qty': 50}],
+            location_id=self.test_stock_location_02.id,
+        )
+        pick.move_lines[0].write({'quantity_done': 10})
+        pick.move_lines[1].write({'quantity_done': 10})
+        product_ids = pick.move_line_ids.mapped('product_id')
+        pick_name = pick.name
+        bk_picking = pick._backorder_move_lines()
+        self.assertEqual(int(pick_name[-5]), int(bk_picking
+        .name[-5]))
+        self.assertEqual(product_ids, bk_picking.move_line_ids.mapped('product_id'))
+        self.assertEqual([50.0, 50.0], bk_picking.move_lines.mapped('product_qty'))
+        self.assertEqual([10.0, 10.0], bk_picking.move_line_ids.mapped('qty_done'))
+
+    def test12_backorder_move_lines_simple(self):
+        """ Backorder move lines for incomplete pick, subset of mls """
+        self.create_quant(self.fig.id, self.test_stock_location_02.id, 50)
+        self.create_quant(self.banana.id, self.test_stock_location_02.id, 50)
+        pick = self.Picking.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{'product': self.fig, 'qty': 50}, {'product': self.banana, 'qty': 50}],
+            location_id=self.test_stock_location_02.id,
+        )
+        pick.move_lines[0].write({'quantity_done': 10})
```

Comment:
There is no need to use `.write()` when updating just one move, can do: `pick.move_lines[0].quantity_done= 10`